rust/scx_stats/scx_stats_derive/src/lib.rs

```diff
@@ -46,3 +47,99 @@ pub fn stat(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
 
     output.into()
 }
+
+#[proc_macro_attribute]
+pub fn stat_doc(
+    _attr: proc_macro::TokenStream,
+    item: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let input = parse_macro_input!(item as DeriveInput);
+    let ident = input.ident;
+    let vis = input.vis;
+    let attrs = input.attrs;
+    let generics = input.generics;
+    let data = input.data;
+
+    let mut output = proc_macro2::TokenStream::new();
+
+    if let Data::Struct(data_struct) = data {
+        let fields = match data_struct.fields {
+            Fields::Named(fields_named) => fields_named.named,
+            _ => {
+                return syn::Error::new_spanned(
+                    ident,
+                    "stat attribute can only be used on structs with named fields",
+                )
+                .to_compile_error()
+                .into();
+            }
+        };
+
+        let mut new_fields = Vec::new();
+
+        for mut field in fields {
+            let mut doc_string = None;
+            let mut new_attrs = Vec::new();
+
+            for attr in field.attrs.clone() {
+                if attr.path().is_ident("stat") {
+                    // Parse the arguments within #[stat(...)]
+                    attr.parse_nested_meta(|meta| {
+                        if meta.path.is_ident("desc") {
+                            // Extract the literal string value from `desc`
+                            let desc_literal: Lit = meta.value()?.parse()?;
+                            if let Lit::Str(lit_str) = desc_literal {
+                                doc_string = Some(lit_str.value());
+                            }
+                        }
+                        Ok(())
+                    })
+                    .unwrap_or_else(|err| {
+                        panic!("Failed to parse the stat attribute: {}", err);
+                    });
+                }
+                new_attrs.push(attr);
+            }
+
+            // If a description string was found, add a #[doc = "..."] attribute
```

Comment:
Can we just rename desc field to doc?

Reply:
changing `#[stat(desc = "...")]` into `#[stat(doc = "...")]` won't make it into the generated documentation because doc attribute is actually `#[doc = "..."]`.
Also, `desc` in `stat` macro here is put into the `StatsMeta`, for example, here is the expanded rust code:
<img width="1488" alt="image" src="https://github.com/user-attachments/assets/0a840d55-0dd7-4834-9e4c-75b30f311eb7">

with the change in this PR, the original struct is extended with document like the following, where `/// ...` lines are added by the `stat_doc` macro.
<img width="556" alt="image" src="https://github.com/user-attachments/assets/089777b5-7e3c-4c25-9872-29489ed6de8b">

Reply:
Hmmm... I wonder whether it should be synced the other way then - ie. let scx_stats macro to take the `///` comment and turn that into description field. What do you think?

Reply:
I think it should be the current way. Documents can be out of rules, while a `desc` property feels constrained to be one short sentence. In the current way, it's also composable as one can add more doc in comment. So more than sync, it's more an extension to the doc from the desc property. Screenshots for a demo are attached: 
<img width="579" alt="image" src="https://github.com/user-attachments/assets/5f7bf867-dba2-4a51-9dd8-4500cf63bca4">

<img width="550" alt="image" src="https://github.com/user-attachments/assets/9174f90b-dd30-430f-a98d-4a9b26356b95">

Reply:
Understood. Thanks for the explanation and the PR!